addons/mail/static/src/models/thread/thread.js

```diff
@@ -740,6 +770,15 @@ function factory(dependencies) {
             return [['replace', this.cache('[]')]];
         }
 
+        /**
+         * @private
+         * @returns {mail.message[]}
+         */
+        _computeMessages() {
+            const allMessages = [...new Set(this.mainCacheMessages.concat(this.originThreadMessages))];
```

Comment:
Set is not needed, relational field is already taking care of removing duplicates now.

Reply:
Hm... you mean `'replace'` already takes care of duplicates? Yes it does, but it's less processing and more expressive to not pass duplicates in the 1st place, so I would prefer keeping this LOC.